recipes/lock.go

```diff
@@ -0,0 +1,124 @@
+package recipes
+
+import (
+	"time"
+
+	"github.com/betable/ezk"
+	"github.com/samuel/go-zookeeper/zk"
+)
+
+type Lock struct {
+	Path   string
+	MaxAge time.Duration
+	client *ezk.Client
+	acl    []zk.ACL
+	lock   string
+}
+
+// NewLock initializes a new distributed lock.
+func NewLock(client *ezk.Client, path string, acl []zk.ACL) *Lock {
+	return &Lock{
+		Path:   path,
+		client: client,
+		acl:    acl,
+	}
+}
+
+// WithCleaner adds a cleaner to the lock, this will remove all the files
+// in the Lock.Path older than t.
+func (l *Lock) WithCleaner(t time.Duration) {
```

Comment:
Could we add a comment for the big picture here. To play devil's advocate, why use a lock with cleaner when the lock is already ephemeral and will disappear when the last client detaches?